tests/unit/connection_parameters_tests.py

```diff
@@ -241,6 +244,11 @@ def test_heartbeat(self):
         with self.assertRaises(ValueError):
             params.heartbeat = -1
 
+        def heartbeat_callback(val):
+            return 1
+        params.heartbeat = heartbeat_callback
+        self.assertTrue(callable(params.heartbeat))
+
```

Comment:
Use `self.assertIs(params.heartbeat, heartbeat_callback)`

Reply:
Also need a ConnectionParameters test for passing callable heartbeat in constructor under ConnectionParametersTests